### PR TITLE
Files structure and auth feature.

### PR DESCRIPTION
**Pull request by ahmed-sala**

### CODE REVIEW

**Comment by john-safwat** [comment]
I appreciate your choice to use the Retrofit package, as it demonstrates a solid understanding of effective API handling. However, I noticed some violations of naming conventions in your code, specifically regarding naming files.

### Naming Conventions in Dart:
- **Class names should be in camel case:** For example, `AuthRepository`
- **File names should be in snake_case:** For example, `auth_repository`
- **Function and variable names should be in camel case, except the first word should be all lowercase:** For example, `signInUserWithEmailAndPassword()`

File: lib/core/error_handeling/AppExceptions.dart [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class ParsingError implements Exception {
```
Comment (omar-dev0) [inline lib/core/error_handeling/AppExceptions.dart:1]: In Dart we already have FormatException class you don't need this

File: lib/features/auth/data/api/model/response/AuthResponse.dart [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import 'package:json_annotation/json_annotation.dart';
+
+import '../../../../domain/model/AppUser.dart';
```
Comment (omar-dev0) [inline lib/features/auth/data/api/model/response/AuthResponse.dart:3]: using full path more safe than relative path 
for some reasons like 
1. Code Refactoring Stability:
2. Consistency Across Modules:
for example -> 
**import 'package:your_project_name/domain/model/AppUser.dart';** 
is more safe than 
**import '../../../../domain/model/AppUser.dart';**

**Comment by ahmed-sala** [comment]
> I appreciate your choice to use the Retrofit package, as it demonstrates a solid understanding of effective API handling. However, I noticed some violations of naming conventions in your code, specifically regarding naming files.
> 
> ### Naming Conventions in Dart:
> * **Class names should be in camel case:** For example, `AuthRepository`
> * **File names should be in snake_case:** For example, `auth_repository`
> * **Function and variable names should be in camel case, except the first word should be all lowercase:** For example, `signInUserWithEmailAndPassword()`

**Comment by ahmed-sala** [comment]
> > I appreciate your choice to use the Retrofit package, as it demonstrates a solid understanding of effective API handling. However, I noticed some violations of naming conventions in your code, specifically regarding naming files.
> > ### Naming Conventions in Dart:
> > 
> > * **Class names should be in camel case:** For example, `AuthRepository`
> > * **File names should be in snake_case:** For example, `auth_repository`
> > * **Function and variable names should be in camel case, except the first word should be all lowercase:** For example, `signInUserWithEmailAndPassword()`

File: lib/core/error_handeling/AppExceptions.dart (hunk 1 above)
Comment (ahmed-sala) [reply lib/core/error_handeling/AppExceptions.dart:1]: Eng Mohamed said this is better.

**Pull request merged**